plugin-server/tests/shared/db.test.ts

```diff
@@ -89,4 +93,82 @@ describe('DB', () => {
             expect(await db.fetchGroupTypes(2)).toEqual({ group_name: 0, foo: 1 })
         })
     })
+
+    describe('updatePersonProperties', () => {
+        //  How we expect this query to behave:
+        //    | value exists | method   | previous method | previous TS is ___ call TS | write/override
+        //  1 | no           | N/A      | N/A             | N/A                        | yes
+        //  2 | yes          | set      | set             | before                     | yes
+        //  3 | yes          | set_once | set             | before                     | no
+        //  4 | yes          | set      | set_once        | before                     | yes
+        //  5 | yes          | set_once | set_once        | before                     | no
+        //  6 | yes          | set      | set             | equal                      | no
+        //  7 | yes          | set_once | set             | equal                      | no
+        //  8 | yes          | set      | set_once        | equal                      | yes
+        //  9 | yes          | set_once | set_once        | equal                      | no
+        // 10 | yes          | set      | set             | after                      | no
+        // 11 | yes          | set_once | set             | after                      | no
+        // 12 | yes          | set      | set_once        | after                      | yes
+        // 13 | yes          | set_once | set_once        | after                      | yes
+
+        let team: Team
+        let person: Person
+        const uuid = new UUIDT().toString()
+        const distinct_id = 'distinct_id_update_person_properties'
+
+        const FUTURE_TIMESTAMP = DateTime.fromISO('2050-10-14T11:42:06.502Z')
+        const PAST_TIMESTAMP = DateTime.fromISO('2000-10-14T11:42:06.502Z')
+
+        beforeEach(async () => {
+            team = await getFirstTeam(hub)
+            person = await db.createPerson(PAST_TIMESTAMP, {}, team.id, null, false, uuid, [distinct_id])
```

Comment:
if I'm reading this correctly, there's a pretty bad pattern in this test. We're "creating a person" each time, but in reality it only gets created once given we never delete it.

Then we don't clear it and keep state over test runs.

Reply:
https://github.com/PostHog/posthog/blob/2929bcdeecb873c4d6e1d705e7e91f4367593dad/plugin-server/tests/shared/db.test.ts#L20 resets the test DB before each test (just double checked by console.logging in both the inner and outer beforeEach that both are always ran)

Reply:
ah yes indeed, forgot that 👍